src/trexio.c

```diff
@@ -183,14 +183,17 @@ trexio_exit_code trexio_read_nucleus_coord(trexio_t* file, double* coord) {
   int64_t dim_coord = nucleus_num*3;
   if (dim_coord < 0) return TREXIO_FAILURE;
 
+  int rank = 2;
+  uint64_t dims[2] = {nucleus_num, 3}; 
+
   switch (file->back_end) {
 
   case TREXIO_TEXT:
     return trexio_text_read_nucleus_coord(file, coord, (uint64_t) dim_coord);
```

Comment:
OK. So I will need to update the text back end to be consistent with what you do (rank+array).

Reply:
I can do it next week if you prefer